server/player/state/player_state.py

```diff
@@ -19,24 +19,50 @@
 # along with taserver.  If not, see <http://www.gnu.org/licenses/>.
 #
 import inspect
+from functools import wraps
+
+from player.player import Player
 
 
 class PlayerState:
-    def __init__(self, player):
+    def __init__(self, player: Player):
         self.player = player
 
     def handle_request(self, request):
         methods = [
             func for name, func in inspect.getmembers(self) if
             not name.startswith("__")
-            and hasattr(func, "handler")
-            and isinstance(request, func.packet)
+            and getattr(func, 'packet', None) == type(request)
```

Comment:
I think it's clearer to the reader when the attribute is named handles_packet or handled_packet_type. Then they can see that we are matching the actual request against an attribute that indicates which packet the function handles.

Reply:
i think the benefits of a readable decorator are more important @handles(packet=...) instead of @handles(handled_packet=... ?)

Reply:
You can leave the parameter name as is. I'm only talking about the member variable.